### DISCUSSION TITLE
Vercel cache is empty

### DISCUSSION THREAD

**Discussion by tomy0000000** [vercel/community#1677] [Help] [answered]
### Summary

Vercel is caching and returning empty responses, while the edge response works fine.

### Additional information

I recently tried the `@vercel/og`. The repository is [tomy0000000/og](https://github.com/tomy0000000/og), and it's deployed on vercel at https://og.tomy.tech

However, I noticed that sometimes vercel caches empty responses.

Here's an example (at the time of writing, you can find more by following the steps in the [Suspect](#suspect) part):
https://og.tomy.tech/api/blog?title=2023%20%E5%B9%B4%201%20%E6%9C%88%E5%A0%B1%E5%A0%B1

```shell
$ curl 'https://og.tomy.tech/api/blog?title=2023%20%E5%B9%B4%201%20%E6%9C%88%E5%A0%B1%E5%A0%B1'
(no output here)
$ 
```

But it's working as expected when I add the `_vercel_no_cache=1` argument.

<img width="1196" alt="CleanShot 2023-03-03 at 12 35 23@2x" src="https://user-images.githubusercontent.com/23290356/222632319-cedff609-7d1f-44ed-9b3c-4fd7a88e34d5.png">

## Suspect

I found this weird behavior while developing [my blog](https://github.com/tomy0000000/og). When I trigger a build, Hugo (the framework engine) will send multiple requests to generate a batch of OG images for the blog in a short time. I'm guessing that some requests failed to generate the response and are cached by vercel.

Steps to reproduce

```shell
git clone https://github.com/tomy0000000/blog.git
hugo server -D
```

Here's an example output: 

```shell
Start building sites … 
hugo v0.104.3+extended darwin/amd64 BuildDate=unknown
ERROR 2023/03/03 12:49:19 render of "term" failed: "/Users/tomy0000000/Library/CloudStorage/Dropbox/Developing/Websites/blog/layouts/_default/rss.xml:67:56": execute of template failed: template: _default/rss.xml:67:56: executing "_default/rss.xml" at <$image.Content>: error calling Content: error calling resources.GetRemote: failed to resolve media type for remote resource "https://og.tomy.tech/api/blog?title=%E6%89%B9%E6%AC%A1%E5%BE%9E%20PDF%20%E8%BD%89%E6%AA%94%E5%87%BA%E6%96%87%E5%AD%97%E4%BE%86"
ERROR 2023/03/03 12:49:19 render of "home" failed: "/Users/tomy0000000/Library/CloudStorage/Dropbox/Developing/Websites/blog/layouts/_default/rss.xml:67:56": execute of template failed: template: _default/rss.xml:67:56: executing "_default/rss.xml" at <$image.Content>: error calling Content: error calling resources.GetRemote: failed to resolve media type for remote resource "https://og.tomy.tech/api/blog?title=%E6%89%B9%E6%AC%A1%E5%BE%9E%20PDF%20%E8%BD%89%E6%AA%94%E5%87%BA%E6%96%87%E5%AD%97%E4%BE%86"
ERROR 2023/03/03 12:49:19 render of "section" failed: "/Users/tomy0000000/Library/CloudStorage/Dropbox/Developing/Websites/blog/layouts/_default/rss.xml:67:56": execute of template failed: template: _default/rss.xml:67:56: executing "_default/rss.xml" at <$image.Content>: error calling Content: error calling resources.GetRemote: failed to resolve media type for remote resource "https://og.tomy.tech/api/blog?title=%E6%89%B9%E6%AC%A1%E5%BE%9E%20PDF%20%E8%BD%89%E6%AA%94%E5%87%BA%E6%96%87%E5%AD%97%E4%BE%86"
Error: Error building site: failed to render pages: render of "term" failed: "/Users/tomy0000000/Library/CloudStorage/Dropbox/Developing/Websites/blog/layouts/_default/rss.xml:67:56": execute of template failed: template: _default/rss.xml:67:56: executing "_default/rss.xml" at <$image.Content>: error calling Content: error calling resources.GetRemote: failed to resolve media type for remote resource "https://og.tomy.tech/api/blog?title=%E6%89%B9%E6%AC%A1%E5%BE%9E%20PDF%20%E8%BD%89%E6%AA%94%E5%87%BA%E6%96%87%E5%AD%97%E4%BE%86"
Built in 209 ms
```

You can find the `failed to resolve media type for remote resource` error, which is caused by the empty cache response from vercel.

**Comment by utkarshkukreti** [comment, marked as answer]
This looks like the same issue as https://github.com/orgs/vercel/discussions/1437.

**Reply by tomy0000000** [reply]
Looks like so, I'll try your patch later on. Thanks for the hint.
